storage/client/internal/parse_rfc3339_test.cc

```diff
@@ -0,0 +1,348 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "storage/client/internal/parse_rfc3339.h"
+#include <gtest/gtest.h>
+#include <ctime>
+
+TEST(ParseRfc3339Test, ParseEpoch) {
+  auto timestamp = storage::internal::ParseRfc3339("1970-01-01T00:00:00Z");
+  // The C++ 11, 14 and 17 standards do not guarantee that the system clock's
+  // epoch is actually the same as the Unix Epoch. Luckily, the platforms we
+  // support actually have that property, and C++ 20 fixes things. If this test
+  // breaks because somebody is unlucky enough to have a C++ compiler + library
+  // with a different Epoch then at least we know what the problem is.
+  EXPECT_EQ(0, timestamp.time_since_epoch().count());
```

Comment:
`EXPECT_EQ` returns an output stream, so you can add a helpful error message here:

```cpp
EXPECT_EQ(0, ...) << "Your C++ compiler and/or library do not set 1970-01-01 as the beginning of Unix time.";
// consider adding a reference to __FILE__ for further info, or add additional context as to what this means: will it break for users, etc.
```

Thoughts?

Reply:
Good idea.